docs/root/configuration/http/http_filters/header_mutation_filter.rst

```diff
@@ -0,0 +1,10 @@
+.. _config_http_filters_header_mutation:
+
+Header Mutation
+===============
+
+* This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.filters.http.header_mutation.v3.HeaderMutation``.
```

Comment:
How should an operator think about this in relation to the early HTTP mutation filter that already exists? Is this a strict superset an basically obviates it?

Reply:
This would be a more flexible alternative to the tranditional `headers_to_add`/`headers_to_remove` in most cases.

However, this still be a normal HTTP filter. In some corner cases, this filter cann't be used as alternative of `headers_to_add`/`headers_to_remove`. For example, if `sendDirectLocalReply` is called, then the encoder filters will be skipped.

Reply:
OK, maybe inline this explanation.